### PR TITLE
Add Entity:getTable() and Entity:getVar()

### PR DESCRIPTION
**Pull request by Some1Lambda**
- Added Entity:getTable()
- Added Entity:getVar()
- Added the entities.getTable permission required by both getTable and getVar functions, default is Can Tool

### CODE REVIEW

**Comment by thegrb93** [comment]
this could be way too intrusive. e.g. grabbing the password of a wire keypads, grabbing code from starfall/e2 chips. The permission helps but still needs to be carefully considered

**Comment by Some1Lambda** [comment]
Maybe we could make it admin only by default, or instead just screw the permission and only allow it on the player's own entities?

**Review by thegrb93 — APPROVED** [review]
I gotta test these

**Comment by thegrb93** [comment]
I think this one will need to be admin/superuser only.

File: lua/starfall/libs_sv/entities.lua [hunk 1]
```diff
@@ -934,4 +935,22 @@ function ents_methods:setPhysicsUpdateListener(func)
 	end
 end
 
+--- Returns entity's table.
```
Comment (thegrb93) [inline lua/starfall/libs_sv/entities.lua:938]:
```suggestion
--- Returns entity's glua table containing internal glua variables about the entity
```

File: lua/starfall/libs_sv/entities.lua [hunk 2]
```diff
@@ -934,4 +935,22 @@ function ents_methods:setPhysicsUpdateListener(func)
...
+-- @return table The entity's table.
+function ents_methods:getTable()
+    local ent = getent(self)
+    checkpermission(instance, ent, "entities.getTable")
+    return instance.Sanitize(ent:GetTable())
+end
+
+--- Returns a variable from the entity's table.
```
Comment (thegrb93) [inline lua/starfall/libs_sv/entities.lua:946]:
```suggestion
--- Returns a variable from the entity's internal glua table.
```

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
I've been procrastinating... will try this asap

**Comment by thegrb93** [comment]
My main worry is using it on a starfall entity and cloning its instance might give access to something really bad or cause something really bad. I'm setting it to admin only for now.

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**